src/core/state.go

```diff
@@ -951,7 +951,7 @@ func (state *BuildState) queueTestTarget(target *BuildTarget) {
 
 // queueResolvedTarget is like queueTarget but once we have a resolved target.
 func (state *BuildState) queueResolvedTarget(target *BuildTarget, forceBuild, neededForSubinclude bool) error {
-	target.NeededForSubinclude = target.NeededForSubinclude || neededForSubinclude
+	target.neededForSubinclude.Or(neededForSubinclude)
```

Comment:
Interesting to see this functional pattern emerge for go sync stuff. Hadn't seen it before.

Reply:
Actually now wondering if I should get rid of Or and just do the if here... that might be simpler